x-pack/plugins/triggers_actions_ui/public/application/sections/rule_details/components/rule_alerts_summary.tsx

```diff
@@ -0,0 +1,196 @@
+                        id="xpack.triggersActionsUI.sections.ruleDetails.rule.ruleSummary.allAlertsTitle"
+                        defaultMessage="All alerts"
+                      />
+                    </EuiText>
+                    <EuiText>
+                      <h4>{active + recovered}</h4>
+                    </EuiText>
+                  </EuiFlexItem>
+                </EuiFlexGroup>
+                <EuiFlexGroup direction="row">
+                  <EuiFlexItem>
+                    <EuiText size="s" color="subdued">
+                      <FormattedMessage
+                        id="xpack.triggersActionsUI.sections.ruleDetails.rule.ruleSummary.activeTitle"
+                        defaultMessage="Active"
+                      />
+                    </EuiText>
+                    <EuiText color="#4A7194">
+                      <h4>{active}</h4>
+                    </EuiText>
+                  </EuiFlexItem>
+                </EuiFlexGroup>
+                <EuiFlexGroup direction="row">
+                  <EuiFlexItem>
+                    <EuiText size="s" color="subdued">
+                      <FormattedMessage
+                        id="xpack.triggersActionsUI.sections.ruleDetails.rule.ruleSummary.recoveredTitle"
+                        defaultMessage="Recovered"
+                      />
+                    </EuiText>
+                    <EuiFlexItem>
+                      <EuiText color="#C4407C">
+                        <h4>{recovered}</h4>
+                      </EuiText>
+                    </EuiFlexItem>
+                  </EuiFlexItem>
+                </EuiFlexGroup>
+              </EuiFlexGroup>
+            </EuiPanel>
+            <EuiHorizontalRule margin="none" />
+          </EuiFlexGroup>
+        </EuiFlexItem>
+
+        <EuiFlexItem grow={false}>
+          <EuiTitle size="xxs">
+            <h5>
+              <FormattedMessage
+                id="xpack.triggersActionsUI.sections.ruleDetails.rule.ruleSummary.recentAlertHistoryTitle"
+                defaultMessage="Recent alert history"
+              />
+            </h5>
+          </EuiTitle>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+      {AlertsChart({ data: alertsChartData })}
+    </EuiPanel>
+  );
+};
+
+// eslint-disable-next-line import/no-default-export
+export { RuleAlertsSummary as default };
+interface AlertsChartProps {
+  data: AlertChartData[];
+}
+const AlertsChart = ({ data }: AlertsChartProps) => {
+  const theme: PartialTheme = {
+    chartMargins: {
+      bottom: 0,
+      left: 0,
+      top: 20,
+      right: 0,
+    },
+    chartPaddings: {
+      bottom: 0,
+      left: 0,
+      top: 0,
+      right: 0,
+    },
+  };
+
+  return (
+    <Chart size={['100%', '35%']}>
+      <Settings tooltip={TooltipType.None} theme={theme} />
+      <BarSeries
+        id="bars"
+        xScaleType={ScaleType.Time}
+        yScaleType={ScaleType.Linear}
+        xAccessor="x"
+        yAccessors={['y']}
```

Comment:
we can create constants for `['x'], ['y'], ['g']`

Reply:
I'm not sure I have got your point. You mean `const X = 'x'` !
Could you please elaborate more on that?